### PR TITLE
Update to go1.21.1

### PR DESCRIPTION
**Pull request by dusk125**
Addresses a to do in https://github.com/etcd-io/etcd/issues/16588

### CODE REVIEW

**Comment by tjungblu** [comment]
/lgtm

finally unsafe.Slice is implemented in go! :tada:

**Review by jmhbnz — APPROVED** [review]
LGTM

**Review by serathius — APPROVED** [review]

File: freelist.go [hunk 1]
```diff
@@ -313,13 +313,13 @@ func (f *freelist) write(p *common.Page) error {
 		p.SetCount(uint16(l))
 		var ids []common.Pgid
 		data := common.UnsafeAdd(unsafe.Pointer(p), unsafe.Sizeof(*p))
-		common.UnsafeSlice(unsafe.Pointer(&ids), data, l)
+		ids = unsafe.Slice((*common.Pgid)(data), l)
```
Comment (ahrtr) [inline freelist.go:316]: minor comment: line 314 can be removed? same below.
```suggestion
		ids := unsafe.Slice((*common.Pgid)(data), l)
```

File: internal/common/page.go [hunk 2]
```diff
@@ -101,7 +101,7 @@ func (p *Page) LeafPageElements() []leafPageElement {
 	}
 	var elems []leafPageElement
 	data := UnsafeAdd(unsafe.Pointer(p), unsafe.Sizeof(*p))
-	UnsafeSlice(unsafe.Pointer(&elems), data, int(p.count))
+	elems = unsafe.Slice((*leafPageElement)(data), int(p.count))
```
Comment (ahrtr) [inline internal/common/page.go:104]: ditto, remove line 102. Same below
```suggestion
	elems := unsafe.Slice((*leafPageElement)(data), int(p.count))
```

File: internal/common/page.go [hunk 3]
```diff
@@ -118,7 +117,7 @@ func (p *Page) BranchPageElements() []branchPageElement {
 	}
 	var elems []branchPageElement
 	data := UnsafeAdd(unsafe.Pointer(p), unsafe.Sizeof(*p))
-	UnsafeSlice(unsafe.Pointer(&elems), data, int(p.count))
+	elems = unsafe.Slice((*branchPageElement)(data), int(p.count))
```
Comment (ahrtr) [inline internal/common/page.go:120]: remove line 118
```suggestion
	elems := unsafe.Slice((*branchPageElement)(data), int(p.count))
```

File: internal/common/page.go [hunk 4]
```diff
@@ -151,7 +150,7 @@ func (p *Page) FreelistPageIds() []Pgid {
 
 	var ids []Pgid
 	data := UnsafeIndex(unsafe.Pointer(p), unsafe.Sizeof(*p), unsafe.Sizeof(ids[0]), idx)
-	UnsafeSlice(unsafe.Pointer(&ids), data, count)
+	ids = unsafe.Slice((*Pgid)(data), count)
```
Comment (ahrtr) [inline internal/common/page.go:153]: remove line 151
```suggestion
	ids := unsafe.Slice((*Pgid)(data), count)
```

**Review by ahrtr — APPROVED** [review]
LGTM

thx

**Pull request merged**